clients/cli/src/ui/dashboard/updaters.rs

```diff
@@ -0,0 +1,248 @@
+//! Dashboard state update logic
+//!
+//! Contains all methods for updating dashboard state from events
+
+use super::state::{DashboardState, FetchingState};
+
+use crate::events::{EventType, Worker};
+use crate::ui::metrics::{SystemMetrics, TaskFetchInfo, ZkVMMetrics};
+
+use std::time::Instant;
+
+impl DashboardState {
+    /// Update the dashboard state with new tick and metrics.
+    pub fn update(&mut self) {
```

Comment:
Is there no way to do this in a more robust, structured way? I hate that we're doing substring searches over text messages sent to the user, rather than just holding the events in a structured format that we can work through appropriately.